cli/cmd/plugin/diagnostics/e2e-test/e2e_test.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e_test
+
+import (
+	"bytes"
+	"crypto/rand"
+	"fmt"
+	"io"
+	"log"
+	"math/big"
+	"os"
+	"os/exec"
+	"testing"
+)
+
+var myDir string
+var tceRepoPath string
+var tanzuDiagnosticPluginDir string
+var tanzuDiagnosticBin string
+var clusterName string
+var clusterKubeContext string
+var home string
+
+const (
+	colorReset = "\033[0m" // Reset
+	colorBlue  = "\033[34m"
+	colorRed   = "\033[31m" // Fail
+	colorGreen = "\033[32m" // Pass
+	passConst  = "GREEN"
+	failConst  = "RED"
+)
+
+func initializeVariable() {
+	// variables used across functions
+	myDir, _ = os.Getwd()
+	tceRepoPath = os.Getenv("ROOT_DIR")
+	tanzuDiagnosticPluginDir = myDir + "/.."
+	tanzuDiagnosticBin = myDir + "/tanzu-diagnostics-e2e-bin"
+	home = os.Getenv("HOME")
+	rand, _ := rand.Int(rand.Reader, big.NewInt(1000))
+	clusterName = "uc" + rand.String() + "test"
+	clusterKubeContext = "kind-" + clusterName
+}
+
+func TestSetup(t *testing.T) {
+	signal := passConst
+	fmt.Println("-------------------------------", string(colorBlue), "Diagnostic e2e Test", string(colorReset), "---------------------------------------------")
+
+	initializeVariable()
+	//Installing TCE
+	err := os.Chdir(tceRepoPath)
+	if err != nil {
+		log.Println("error while changing directory", err)
+		signal = failConst
+	}
+	err = runDeployScript("test/download-or-build-tce.sh")
```

Comment:
During the "Check - Diagnostics Plugin" workflow, this causes a full build to be run twice. We should either remove the full build as a separate step from there, or add a check here first to see if `tanzu` is already present.

But that said, it looks like now we are only calling `go run` to test the code, so I don't see why we need a full install. So can we just remove this?

Reply:
+1

Reply:
+1

Reply:
There is check for `tanzu` cli in diagnostic code [here](https://github.com/vmware-tanzu/community-edition/blob/aa4549af85e8073ddec564ed9a9025b9831de696/cli/cmd/plugin/diagnostics/pkg/collect_cmd.go#:~:text=gexe.Prog().Avail(%22-,tanzu,-%22)%20%3D%3D%20%22%22%20%7B) due to which we need tanzu.

Reply:
I think we should remove that `tanzu` binary check that you referenced.

We should actually do the second option:
`remove the full build as a separate step from there` as it is the only way for us to trigger and consume from the nightly build

Reply:
Hey @dvonthenen just to clarify you mean removing the tanzu build code from the e2e-test and adding it in workflow? like [this](https://github.com/vmware-tanzu/community-edition/blob/main/.github/workflows/check-pr-diagnostics-plugin.yaml#:~:text=instance%2Did%3A%3A%24%7BINSTANCE_NAME%7D-,build%2Drelease%3A,-name%3A%20Build%20Release)?

Reply:
Yea... I can see how that is confusing. We should remove the build from the workflow and the download-or-build-sh script should do the build.